### PR TITLE
Chore/fnd v5 upgrade

### PR DESCRIPTION
**Pull request by dhruvTor**
Error messages fix

### CODE REVIEW

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift [hunk 1]
```diff
@@ -121,7 +121,7 @@ extension TorusUtils {
                     let msg: String = data["message"] as? String,
                     let ret = BigUInt(msg, radix: 16)
                     else {
-                throw TorusUtilError.decodingFailed
+                throw TorusUtilError.decodingFailed("Message value not correct or nil in \(data["message"])")
```
Comment (michaellee8) [inline Sources/TorusUtils/Extensions/TorusUtils+extension.swift:124]: Can u log the full `data` here, the wider the context is better.

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift [hunk 2]
```diff
@@ -175,14 +175,14 @@ extension TorusUtils {
...
-                    throw TorusUtilError.decodingFailed
+                    throw TorusUtilError.decodingFailed(decoded.error?.data)
                 }
                 os_log("retrieveDecryptAndReconstuct: %@", log: getTorusLogger(log: TorusUtilsLogger.core, type: .info), type: .info, "\(decoded)")
                 
                 guard
                     let decodedResult = decoded.result as? [String:Any],
                     let keyObj = decodedResult["keys"] as? [[String:Any]]
-                else { throw TorusUtilError.decodingFailed }
+                else { throw TorusUtilError.decodingFailed("keys not found in result \(decoded.result)")}
```
Comment (michaellee8) [inline Sources/TorusUtils/Extensions/TorusUtils+extension.swift:185]: Maybe just log full decoded, here sometimes maybe result is empty.

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift [hunk 3]
```diff
@@ -306,7 +306,7 @@ extension TorusUtils {
                         guard
                             let r = a.result as? [String:String]
                         else {
-                            throw TorusUtilError.decodingFailed
+                            throw TorusUtilError.decodingFailed("\(a.result) not found")
```
Comment (michaellee8) [inline Sources/TorusUtils/Extensions/TorusUtils+extension.swift:309]: again full `a`

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift [hunk 4]
```diff
@@ -625,17 +627,14 @@ extension TorusUtils {
                 guard
                     let index = newEndpoints2.firstIndex(of: endpoint)
                 else {
-                    seal.reject(TorusUtilError.decodingFailed)
+                    seal.reject(TorusUtilError.decodingFailed(nil))
```
Comment (michaellee8) [inline Sources/TorusUtils/Extensions/TorusUtils+extension.swift:630]: Try to log something please, maybe log newEndpoints2

**Review by michaellee8 — CHANGES_REQUESTED** [review]
@dhruv500 Please address the comments. Basically log more things, log the wider context, and don't use nil for log message, log some actual text insted. User's device may not be able to provider stacktrace and our clients won't necessarily have the time to look at those errors.

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift (hunk 1 above)
Comment (dhruvTor) [reply Sources/TorusUtils/Extensions/TorusUtils+extension.swift:124]: Hey @michaellee8 if you see on line 119 we are logging the data

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift (hunk 2 above)
Comment (dhruvTor) [reply Sources/TorusUtils/Extensions/TorusUtils+extension.swift:185]: on line 180 we are logging decoded

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift (hunk 3 above)
Comment (dhruvTor) [reply Sources/TorusUtils/Extensions/TorusUtils+extension.swift:309]: Done

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift (hunk 4 above)
Comment (dhruvTor) [reply Sources/TorusUtils/Extensions/TorusUtils+extension.swift:630]: Done

File: Sources/TorusUtils/Extensions/TorusUtils+extension.swift (hunk 1 above)
Comment (michaellee8) [reply Sources/TorusUtils/Extensions/TorusUtils+extension.swift:124]: i meant embeding those in the errors. Sometimes the client don;t use info level logs on prod

Comment (dhruvTor) [reply Sources/TorusUtils/Extensions/TorusUtils+extension.swift:124]: Got it

File: Package.swift [hunk 5]
```diff
@@ -14,7 +14,7 @@ let package = Package(
             targets: ["TorusUtils"]),
     ],
     dependencies: [
-        .package(name:"FetchNodeDetails", url: "https://github.com/torusresearch/fetch-node-details-swift", from: "1.4.0"),
+        .package(name: "FetchNodeDetails", url: "https://github.com/torusresearch/fetch-node-details-swift.git", .branch("feat/upgradeToVersion5")),
```
Comment (michaellee8) [inline Package.swift:17]: pls rmb to fix this before merging to master

**Review by michaellee8 — APPROVED** [review]

**Comment by michaellee8** [comment]
lgtm except reverting https://github.com/torusresearch/torus-utils-swift/pull/26/files#diff-f913940c58e8744a2af1c68b909bb6383e49007e6c5a12fb03104a9006ae677eR17 before merge to master. @chaitanyapotti pls review

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**